src/main/java/pl/akademia/api/promotion/PromotionCodeRestController.java

```diff
@@ -23,20 +25,30 @@ public PromotionCodeRestController(PromotionCodeService promotionCodeService) {
         return new ResponseEntity<>(promotionCodeService.createPromotionCode(activeDays, discount), HttpStatus.CREATED);
     }
 
-    @GetMapping("/promocodes/{a}")
-    public ResponseEntity<List<PromotionCode>> getAllPromotionCodes(@RequestParam(required = false) String a){
-        List<PromotionCode> promotionCodes;
-        if ("active".equals(a)) promotionCodes = promotionCodeService.getActivePromotionCodeByCode();
-        else if ("inactive".equals(a)) promotionCodes = promotionCodeService.getInactivePromotionCodeByCode();
-        else if (a == null) promotionCodes = promotionCodeService.getAllPromotionCode();
-        else return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-        if (promotionCodes.isEmpty()) return new ResponseEntity<>(HttpStatus.NO_CONTENT);
-        return new ResponseEntity<>(promotionCodes, HttpStatus.OK);
+    @GetMapping("/promocodes/{promoCodeStatus}")
+    public ResponseEntity<List<PromotionCode>> getAllPromotionCodes(@RequestParam(required = false) String promoCodeStatus){
+
+        List<PromotionCode> promotionCodes = null;
+        if (promotionCodes.isEmpty()) {
+            throw new PromoCodeNotFoundException("Promo code  does not exists! ");
+        }
+        if ("active".equals(promoCodeStatus)) {
+            promotionCodes = promotionCodeService.getActivePromotionCodeByCode();
+            return new ResponseEntity<>(promotionCodes, HttpStatus.OK);}
+
+        if ("inactive".equals(promoCodeStatus)){
+            promotionCodes = promotionCodeService.getInactivePromotionCodeByCode();
```

Comment:
a co jeśli promotionCodes będzie pustą listą?

Reply:
masz rację, ale siedzę nad tym i coś nie łapie mi wyjątków, trochę się zakręciłem w tym - jak masz chęć i chwile to poprawić  to się nie gniewam :)